frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,356 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Helmet } from 'react-helmet';
+import { match } from 'react-router-dom';
+import { ClusterServiceVersionLogo, iconFor } from './index';
+import {
+  Alert,
+  Bullseye,
+  Button,
+  Card,
+  CardBody,
+  Flex,
+  FlexItem,
+  FlexModifiers,
+  Spinner,
+} from '@patternfly/react-core';
+import { Firehose, FirehoseResourcesResult } from '@console/internal/components/utils';
+import { k8sUpdate, referenceForModel } from '@console/internal/module/k8s';
+import {
+  ClusterServiceVersionModel,
+  InstallPlanModel,
+  PackageManifestModel,
+  SubscriptionModel,
+} from '../models';
+import {
+  CRDDescription,
+  ClusterServiceVersionKind,
+  SubscriptionKind,
+  InstallPlanKind,
+  PackageManifestKind,
+} from '../types';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
+import { InstallPlanPreview } from './install-plan';
+
+const csvResources = (namespace: string) => [
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    namespaced: true,
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace,
+    prop: 'clusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace: 'openshift-operators',
+    prop: 'allClusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(SubscriptionModel),
+    namespaced: false,
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    isList: true,
+    optional: true,
+    prop: 'subscription',
+  },
+  {
+    kind: referenceForModel(InstallPlanModel),
+    prop: 'installPlan',
+    namespaced: false,
+    isList: true,
+    optional: true,
+  },
+  {
+    isList: true,
+    kind: referenceForModel(PackageManifestModel),
+    namespace: new URLSearchParams(window.location.search).get('catalogNamespace'),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    selector: {
+      matchLabels: {
+        catalog: new URLSearchParams(window.location.search).get('catalog'),
+      },
+    },
+    prop: 'packageManifest',
+  },
+];
+
+const OperatorInstallPageWithoutFirehose: React.FC<OperatorInstallPageProps> = (props) => {
```

Comment:
I think a pattern you see for this tends to be:

```typescript 
const OperatorInstallPage_: React.FC< > ....
```
And wrap `OperatorInstallPage_` inside `OperatorInstallPage`. 


```